### PR TITLE
feat: Added card view for instruments section

### LINKED ISSUE #877: UI of instrument section needs improvement
**Actual Behaviour**

UI of the instrument section part is not up to the mark

**Expected Behaviour**

The UI of the instrument section should have CardViews to represent each instrument.

**Steps to reproduce it**

Change the UI in the code.

**LogCat for the issue**

Not applicable

**Screenshots of the issue**

Not applicable

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by harsh-2711**
Fixes #877 

Changes: Added Card View in instruments section with 1 column in portrait mode and 2 columns in landscape mode. Also great images are added as background.

Screenshots for the change: 
**Portrait Mode**

screenshot_20180518-044330

**Landscape Mode**

screenshot_20180518-102344

APK for testing: 
[app-debug.zip](https://github.com/fossasia/pslab-android/files/2015627/app-debug.zip)

### CODE REVIEW

**Comment by harsh-2711** [comment]
@CloudyPadmal  I have added intents as per my knowledge... Suggest me if anything's wrong.

File: app/src/main/res/layout/application_list_item.xml [hunk 1]
```diff
@@ -8,39 +8,17 @@
...
         android:id="@+id/card_view"
         android:layout_width="match_parent"
-        android:layout_height="match_parent"
+        android:layout_height="@dimen/image_height"
         android:layout_gravity="center"
         android:layout_margin="@dimen/card_margin"
         card_view:cardCornerRadius="@dimen/card_radius">
 
-        <LinearLayout
+        <ImageView
```
Comment (Avjeet) [inline app/src/main/res/layout/application_list_item.xml]: I suggest don't copy the complete image in the card view else separate the text from the background and write the title and description in two textview with an image in the background. That way we will have similar font in all the card views.

**Comment by Avjeet** [comment]
Also add the wave generator card.

**Comment by harsh-2711** [comment]
@Avjeet  I have added wave generator card and also power source card. I have only shared one snapshot. Remaining instruments are also included as decided. @abhinavraj23 did the designing and so I will ask him to do requested changes

**Comment by Vikum94** [comment]
Designs concepts are good. In the oscilloscope card, I think there is an unwanted part of a sine wave overlapping with the larger sine wave.

**Comment by akarshan96** [comment]
@harsh-2711 I think the text shouldn't be the part of the imag. You should add a textview.

**Comment by akarshan96** [comment]
screen shot 2018-05-19 at 11 54 40 am

The card views are implemented in this way in most of the Android apps. We can follow something like this.

**Comment by harsh-2711** [comment]
Finally made the following UI for Instruments section. None of the text is hard-coded..

screenshot_20180520-040849
screenshot_20180520-040932

On clicking how it works should slide the layout down and open another card like view which will give more information about the instrument and on clicking again it will collapse. These changes will be done in subsequent PR of mine

**Comment by Avjeet** [comment]
I think instead of "how it works" you should keep the description where it was earlier like this:
image
as the background images are designed to contain Title and brief description inthe Upper left corner and also It would be one less click for the user.

**Comment by harsh-2711** [comment]
But it fills the whole card and it doesn't looks good.. I tried that but than I changed it like this

**Comment by Vikum94** [comment]
I also agree with @Avjeet Necessarily there is an empty space in the upper left corner. Text suits there.

**Comment by Avjeet** [comment]
@harsh-2711 Could you try by making text size smaller and increasing the height of card a little to accommodate the text or maybe try making description compact.

**Comment by harsh-2711** [comment]
Added descriptions for each instrument as shown
screenshot_20180521-005037

**Comment by harsh-2711** [comment]
Here is the apk for testing
[app-debug.zip](https://github.com/fossasia/pslab-android/files/2020643/app-debug.zip)

**Comment by harsh-2711** [comment]
In my last commit, I changed the description of power source to "Displays value returned by sensors connected to I2C input"

**Comment by CloudyPadmal** [comment]
@harsh-2711 , Why is there a `Learn More` button? We can have the simple card view without much complications.

**Comment by CloudyPadmal** [comment]
Isn't the idea of those 4 vertical lines to align at a same row? Some images are too thick and big while some are small and some are thin. @harsh-2711 @Avjeet @abhinavraj23 @Vikum94 please have identical images and styling. They are svg images. You can edit those right?

**Comment by mariobehling** [comment]
Please post all device and text here (as text). I am unclear what the instrument "sensor" is. This tab is for "instruments". So, an instrument could be a "Lux Meter". Please list all instruments and description with their text.

Please take out "Learn more". This is not documentation to learn more. It is simply a button to access the instrument.

**Comment by harsh-2711** [comment]
Finally made all the svg files from scratch and here is the final output
screenshot_20180523-042738

And here is the apk to test : [app-debug.zip](https://github.com/fossasia/pslab-android/files/2028884/app-debug.zip)

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -639,4 +643,13 @@
...
     <string name="search_menu_title">Search</string>
 
+    <string name="learn_more">Learn More</string>
+
+    <string name="oscilloscope_description">Allows observation of varying \nsignal voltages</string>
+    <string name="multimeter_description">Measure voltages, currents \nand resistances</string>
+    <string name="logic_analyzer_description">Captures and displays signals \nfrom digital systems</string>
+    <string name="sensors_description">Allows logging of data returned \nby sensor connected</string>
+    <string name="wave_generator_description">Generates different waveforms \nand PWM signals</string>
+    <string name="power_source_description">Displays value returned by \nsensors connected to \n I2C input</string>
```
Comment (Avjeet) [inline app/src/main/res/values/strings.xml]: Remove the space before "I2C"

Comment (Avjeet) [reply app/src/main/res/values/strings.xml]: And don't manually create new line using \n instead fix the size of the text view.

**Comment by harsh-2711** [comment]
@Avjeet  Changes done.

**Comment by abhinavraj23** [comment]
@CloudyPadmal @Vikum94 @Avjeet Please review this PR.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -16,12 +16,16 @@
 
 
     <string name="oscilloscope">Oscilloscope</string>
-    <string name="control">Control</string>
-    <string name="advance_control">Advance Control</string>
+   <!-- <string name="control">Control</string> -->
```
Comment (CloudyPadmal) [inline app/src/main/res/values/strings.xml]: If these texts are not in use, please remove them

**Comment by CloudyPadmal** [comment]
This conversation is getting longer. If you have a look at these two tiles,

| Multimeter | Power Source  |
| ----- | ------ |
| screen2 | screen1 |

Do they look nice? It looks like some random shapes here and there

**Comment by harsh-2711** [comment]
@CloudyPadmal  You can suggest some better alternative as I tried to use some +/- text and all that stuff but Android Studio currently doesn't support every feature of Inkscape and thus we can only use the shapes. Should I keep a voltage source along with resistor?

**Comment by CloudyPadmal** [comment]
@harsh-2711 I'm referring to their alignment. Shouldn't they be same?

**Comment by harsh-2711** [comment]
Aligned images

screenshot_20180525-222046__01
screenshot_20180525-223045__01

**Pull request merged**